### ISSUE TITLE
refactor auto-config scripts 

### ISSUE THREAD

**Issue by gvagenas** [RestComm/Restcomm-Connect#917]
Currently the auto-config scripts are unnecessary complicated, with duplicate functionality and several functional errors.
We need to refactor the scripts and maybe come up with a better way to configure Restcomm.

**Comment by hrosa** [comment]
@gvagenas I think RestComm config file has grown quite large in small amount of time. I think best way to start is to review current config options and then check what needs to be implemented via auto-config.

**Comment by deruelle** [comment]
tied to #819  and #399

**Comment by gvagenas** [comment]
@hrosa agree, the root of the problem for the config scripts maze we have here is the complicated Restcomm configuration itself.
@deruelle we should review and cleanup all configuration options and then move to DB

**Comment by hrosa** [comment]
Another thing to take into account: auto-config scripts rely on sed command which is not a proper tool for this task since configuration files are XML based. Given that sed is all about regex, it's common that auto-config break if configuration files suffer minor changes (eclipse auto-format, adding or switching xml nodes, etc).

A much more resilient solution is to use a proper XML lib to read the config files and update the nodes' values. A list of command-line tools that can be called from shell scripts include:
- 4xpath - command-line wrapper around Python's 4Suite package
- XMLStarlet
- xpath - command-line wrapper around Perl's XPath library
- xmllint and xsltproc with little XSL transform scripts to do XML processing from the command line or in shell scripts.

**Comment by gvagenas** [comment]
Ruby, Perl and XSL are also candidates for the tool here
